modules/gcp/centos-gfx/centos-gfx-startup.sh.tmpl

```diff
@@ -13,11 +13,36 @@ DECRYPT_URI="https://cloudkms.googleapis.com/v1/${kms_cryptokey_id}:decrypt"
 PCOIP_AGENT_REPO_PUBKEY_URL=${pcoip_agent_repo_pubkey_url}
 PCOIP_AGENT_REPO_URL=${pcoip_agent_repo_url}
 
+ENABLE_AUTO_SHUTDOWN=${enable_workstation_idle_shutdown}
+AUTO_SHUTDOWN_IDLE_TIMER=${minutes_idle_before_shutdown}
+CPU_POLLING_INTERVAL=${minutes_cpu_polling_interval}
+
 log() {
     local message="$1"
     echo "[$(date)] $${message}" | tee -a "$INST_LOG_FILE"
 }
 
+retry() {
+    local retries=0
+    local max_retries=3
+    until [[ $retries -ge $max_retries ]]
+    do  
+    # Break if command succeeds, or log the retry if command fails.
+        $@ && break || {
+
+            log "--> Failed to run command. $${@}"
+            log "--> Retries left... $(( $max_retries - $retries ))"
+            ((n++))
+            sleep 10;
+        }
+    done
+
+    if [[ $retries -eq $max_retries ]]
+    then
+        return 1
+    fi
+}
```

Comment:
Some of the commands to install idle timer needed retry logic as it sometimes failed. I will be creating a separate commit in the future to refactor any retry logic which will reuse this function.